src/public/components/router.js

```diff
@@ -1,6 +1,70 @@
-import K, * as U from "karet.util"
-import React     from "karet"
+import * as R     from "ramda"
+import * as L     from "partial.lenses"
+import K, * as U  from "karet.util"
+import React      from "karet"
+import toRegex    from 'path-to-regexp'
 
-export default U.withContext(({routes, NotFound}, {path}) => U.fromKefir(K(path, path =>
-  React.createElement(routes[path] || NotFound)
-)))
+// TODO: Make it possible to add routes more than once?
+
+// begin routes initialization
+const expandRoutes = rawRoutes =>
+  R.pipe( R.toPairs
+        , R.map( R.zipObj( [ 'route', 'component' ] ) )
+        )( rawRoutes )
+
+const addRegexes = expandedRoutes =>
+  R.map( route => R.pipe( L.get( 'route' )
+                        , toRegex
+                        , L.set( 'regex' )
+                        )( route )( route )
+       )( expandedRoutes )
+
+const partitionDynamicAndStatic = routesWithRegexes =>
+  R.partition( R.pipe( L.get( [ 'regex', 'keys' ] )
+                     , R.length
+                     , R.lt( 0 )
+                     )
```

Comment:
👍 Agreed, yours is certainly easier to read. I'm beginning to think that I may have drunk too much of the ramda kool-aid.